test/terra/extensions/__init__.py

```diff
@@ -0,0 +1,32 @@
+# This code is part of Qiskit.
```

Comment:
What's the concept of `extensions` in this context?

Reply:
Just copying terra/extensions which is where custom gates and instructions live there